Python/modsupport.c

```diff
@@ -318,32 +342,41 @@ do_mkvalue(const char **p_format, va_list *p_va)
         }
 
         case 'L':
-            return PyLong_FromLongLong((long long)va_arg(*p_va, long long));
+        {
+            long long v = va_arg(*p_va, long long);
+            return PyLong_FromLongLong(v);
+        }
 
         case 'K':
-            return PyLong_FromUnsignedLongLong((long long)va_arg(*p_va, unsigned long long));
+        {
+            unsigned long long v = va_arg(*p_va, unsigned long long);
+            return PyLong_FromUnsignedLongLong(v);
+        }
 
         case 'u':
         {
             PyObject *v;
-            const wchar_t *u = va_arg(*p_va, wchar_t*);
-            Py_ssize_t n;
+            const wchar_t *u = va_arg(*p_va, const wchar_t *);
+            Py_ssize_t n = -1;
             if (**p_format == '#') {
                 ++*p_format;
                 n = va_arg(*p_va, Py_ssize_t);
             }
-            else
-                n = -1;
             if (u == NULL) {
                 v = Py_NewRef(Py_None);
             }
             else {
-                if (n < 0)
-                    n = wcslen(u);
+                if (n < 0) {
+                    n = safe_wcslen(u);
```

Comment:
There is no real issue here. The size of any memory array is limited to `SIZE_MAX` bytes, `wchar_t` is at least 2 bytes, so the result of `wcslen()` does not exceed `SSIZE_MAX = SIZE_MAX/2`.

Reply:
Ah! right! my bad.